### PR TITLE
ui: Add Support for Higher UI Scaling

### PR DESCRIPTION
**Pull request by MasonT8198**
This is so on bigger 4K TVs on a HTPC setup (or just bigger 4K TVs in general) are easier to see in xemu, because even at 2x UI scaling it is hard to see, I have good vision so I could make it out rather easily, but I could see someone without good vision being able to spot it.

### CODE REVIEW

**Comment by MasonT8198** [comment]
Second Add-In, sorry for the PR spam

**Comment by dracc** [comment]
> Second Add-In, sorry for the PR spam

Do not worry about that, PRs touching five different things *are* worse.

Re changes: nit: bad commit titles (try prepending `ui: ` and modifying the rest of them to fit).

**Comment by Aerocatia** [comment]
Perhaps it would be better to not have a hard-coded size menu for this but instead draw at a consistent scale that increases with resolution much like game UI's do.

**Comment by MasonT8198** [comment]
> Perhaps it would be better to not have a hard-coded size menu for this but instead draw at a consistent scale that increases with resolution much like game UI's do.

I could see this being a feature implemented in the feature, I just appended the changes onto the already existing system, as when I use a bigger sized TV, 2x doesn't really help much, granted for me I can see it, but my dad couldn't or my stepmom, so I figured the feature would be a nice addition to an already existing system

File: ui/xemu-hud.cc [hunk 1]
```diff
@@ -1733,14 +1733,14 @@ static void ShowMainMenu()
         {
             int ui_scale_combo = g_ui_scale - 1.0;
             if (ui_scale_combo < 0) ui_scale_combo = 0;
-            if (ui_scale_combo > 1) ui_scale_combo = 1;
-            if (ImGui::Combo("UI Scale", &ui_scale_combo, "1x\0" "2x\0")) {
+            if (ui_scale_combo > 1) ui_scale_combo = ui_scale_combo;
```
Comment (mborgerson) [inline ui/xemu-hud.cc]: This statement now has no effect

File: ui/xemu-hud.cc [hunk 2]
```diff
@@ -1733,14 +1733,14 @@ static void ShowMainMenu()
...
+            if (ImGui::Combo("UI Scale", &ui_scale_combo, "1x\0" "2x\0" "3x\0" "4x\0")) {
                 g_ui_scale = ui_scale_combo + 1;
                 xemu_settings_set_int(XEMU_SETTINGS_DISPLAY_UI_SCALE, g_ui_scale);
                 xemu_settings_save();
                 g_trigger_style_update = true;
             }
-
```
Comment (mborgerson) [inline ui/xemu-hud.cc]: Please restore blank line

In general when submitting patches please try to follow the existing style

File: ui/xemu-hud.cc [hunk 3]
```diff
@@ -1733,14 +1733,14 @@ static void ShowMainMenu()
...
+            ImGui::SameLine(); HelpMarker("Controls how the UI is scaled, not Surface Scaling");
```
Comment (mborgerson) [inline ui/xemu-hud.cc]: I'm not sure this is necessary

**Review by mborgerson — CHANGES_REQUESTED** [review]
There is also a check in `xemu_hud_init` that would need to be updated

**Comment by mborgerson** [comment]
> Perhaps it would be better to not have a hard-coded size menu for this but instead draw at a consistent scale that increases with resolution much like game UI's do.

@Vaporeon  I welcome a more intuitive/automatic scaling method. Can you prototype this?

File: ui/xemu-hud.cc [hunk 4]
```diff
@@ -1732,24 +1732,24 @@ static void ShowMainMenu()
...
-            if (ui_scale_combo > 1) ui_scale_combo = 1;
-            if (ImGui::Combo("UI Scale", &ui_scale_combo, "1x\0" "2x\0")) {
+            if (ImGui::Combo("UI Scale", &ui_scale_combo, "1x\0" "2x\0" "3x\0" "4x\0")) {
                 g_ui_scale = ui_scale_combo + 1;
                 xemu_settings_set_int(XEMU_SETTINGS_DISPLAY_UI_SCALE, g_ui_scale);
                 xemu_settings_save();
                 g_trigger_style_update = true;
             }
-
+
```
Comment (mborgerson) [inline ui/xemu-hud.cc]: Please avoid whitespace changes like this a 2 more below.

File: ui/xemu-hud.cc [hunk 5]
```diff
@@ -1732,24 +1732,24 @@ static void ShowMainMenu()
         if (ImGui::BeginMenu("View"))
         {
             int ui_scale_combo = g_ui_scale - 1.0;
-            if (ui_scale_combo < 0) ui_scale_combo = 0;
-            if (ui_scale_combo > 1) ui_scale_combo = 1;
-            if (ImGui::Combo("UI Scale", &ui_scale_combo, "1x\0" "2x\0")) {
+            if (ImGui::Combo("UI Scale", &ui_scale_combo, "1x\0" "2x\0" "3x\0" "4x\0")) {
```
Comment (mborgerson) [inline ui/xemu-hud.cc:2107]: Don't forget to fix bounds checking when we initialize g_ui_scale at startup

**Comment by MasonT8198** [comment]
@mborgerson got around to fixing up this one, it's been a while, so something may have been done incorrectly.

**Comment by MasonT8198** [comment]
@mborgerson

**Comment by MasonT8198** [comment]
This one has gone stale, I think it's clean now however I am closing this

**Pull request closed without merging**